### PR TITLE
personalization dedicated to the particular surface

### PR DESCRIPTION
**Pull request by shandilya3**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by shandilya3** [comment]
> As we discussed yesterday, rather than filter out the irrelevant propositions ( which totally works), i'd like this sample to change so that the response JSON object is correctly serialized. The ruleset items should not impact the hybrid sample (i don't think). This sample needs to illustrate how to correctly serialize an edge response and provide it to alloy using `applyResponse`. The problem is that response does not seeem to be properly serialized/escaped when putting it into the HTML.
> 
> One other addition to make as part of this PR is to set the `data` property to include the `in-app-response-format`. This will tell XAS to use the new response format. Which very well may resolve this issue altogether 🤞 .
> 
> Set `data` on the event to...
> 
> ```json
> {
>   "__adobe": {
>     "ajo": {
>       "in-app-response-format": 2
>     }
>   }
> }
> ```

```
{
  "__adobe": {
    "ajo": {
      "in-app-response-format": 2
    }
  }
}
``` 
Above code snippet addition didn't help, had to sanitize the html data.  `DOMPurify` takes string full of dirty HTML and it returns a string with clean HTML.

File: ajo/personalization-hybrid/src/server.js [hunk 1]
```diff
@@ -65,6 +69,18 @@ function prepareTemplateVariables(
...
 ) {
   const { headers = {}, body = { handle: [] } } = response;
+  const res =   response.body.handle.filter(
+    (item) => item.type === TYPE_PERSONALIZATION
+  )
+  const sanitizedPayload = res[0].payload.map((item) => ({
+    ...item,
+    items: item.items.map((subItem) => ({
+      ...subItem,
+      data: subItem.data ? sanitizeData(subItem.data) : null,
```
Comment (jasonwaters) [inline ajo/personalization-hybrid/src/server.js:79]: is this really necessary?  I don't think it's reasonable to expect our customers to go to this trouble.

Can you explain the underlying problem in detail and how dompurify is solving it?

File: common/personalization.js [hunk 2]
```diff
@@ -101,6 +108,13 @@ function sendDisplayEvent(aepEdgeClient, req, propositions, cookieEntries) {
             },
           },
         },
+        data : {
```
Comment (jasonwaters) [inline common/personalization.js:111]: `in-app-response-format` is only needed for the requests to get personalization.  not needed for display and interact events.

File: ajo/personalization-hybrid/public/script.js [hunk 3]
```diff
@@ -46,6 +46,13 @@ function sendDisplayEvent(proposition) {
         },
       },
     },
+    data: {
```
Comment (jasonwaters) [inline ajo/personalization-hybrid/public/script.js:49]: `in-app-response-format` is only needed for the requests to get personalization.  not needed for display and interact events.

**Review by jasonwaters — CHANGES_REQUESTED** [review]

File: ajo/personalization-hybrid/src/server.js (hunk 1 above)
Comment (shandilya3) [reply ajo/personalization-hybrid/src/server.js:79]: When comparing the sanitized content with the original, I've noticed differences in the way certain characters are escaped. It's expected that DOMPurify, being a sanitizer, should handle escaping issues.

**Comment by jasonwaters** [comment]
@shandilya3 

please remove `jsdom` and `dompurify`.  We don't need to rely on these third party libs for this.

The work required for this is to get crystal clear on what characters are not being properly escaped.  I decided to do a little more due diligence while writing up this comment.  here is what i learned.

There are four things that must be done when embedding html within a JSON object ( see [here](https://thorntech.com/4-things-you-must-do-when-putting-html-in-json/) to read more about it).

the thing we are not doing from that article is: Escape the forward slash in HTML end tags. 

this 👇 
```html
<div>Hello World!</div><script></script>
```

should be 👇 
```
<div>Hello World!<\/div><script><\/script>
```

when properly escaped like this the JSON strings are parsed properly by browsers.

So, in `server.js` where the applyResponseParam is set to JSON.stringify({...}) you can pass that string into a new function to escape it using something like `stringvalue.replace(/\//g, "\\/")`.

I also noticed that the in-app response looked the same regardless of setting `in-app-response-format` or not.  After chatting with William le, i learned that implementationDetails must be set.  In other words, so long as implementationDetails is set to something, `in-app-response-format` defaults to 2.  so it does not need to be set.  So, you can remove where you set `in-app-response-format` to 2 and instead set implementationDetails to something like this:

```
      implementationDetails: {
        name: "server-side",
      },
```

Finally, although we can (and should) solve the escaping of forward slashes in this sample repo, I think you also need to find out why the forward slash was not escaped in the first place and ensure it gets escaped properly server side.  Maybe it's in XAS?

**Pull request closed without merging**